src/signal.rs

```diff
@@ -474,4 +475,55 @@ mod tests {
         // forever as the loop keeps running. Since we don't join, the thread will become detached
         // as the handle is dropped, and will be killed when the process/main thread exits.
     }
+
+    #[test]
+    fn test_block_unblock_signal() {
+        let signal = SIGRTMIN() + 0;
+
+        // Check if it is blocked.
+        unsafe {
+            let mut sigset: sigset_t = mem::zeroed();
+            pthread_sigmask(SIG_BLOCK, null(), &mut sigset as *mut sigset_t);
+            assert_eq!(sigismember(&sigset, signal), 0);
+        }
+
+        block_signal(signal).unwrap();
+        assert!(get_blocked_signals().unwrap().contains(&(signal)));
+
+        unblock_signal(signal).unwrap();
+        assert!(get_blocked_signals().unwrap().is_empty());
```

Comment:
I think it might be a bit better to do an assertion like `assert!(!get_blocked_signals().unwrap().contains(&(signal)));` (in other words assert the set of blocked signals does not contain `signal`). Checking if it's totally empty might race with other tests that also block signals (like `test_clear_signal`).

Reply:
Will be addressed.